src/components/utils/validation.js

```diff
@@ -88,7 +88,7 @@ export function isVmNameValid (nameCandidate: string): boolean {
 }
 
 export function isDiskNameValid (nameCandidate: string): boolean {
-  return isVmNameValid(nameCandidate)
+  return /^[a-zA-Z0-9_\-\\.]{1,255}$/.test(nameCandidate)
```

Comment:
Please add an unit test  - previously `isVmNameValid()` was unit tested.
@sgratch  @sjd78 
Validation is now more restrictive. However `isDiskNameValid` is also used to validate alias name in `DiskImageEditor`. Can we break some use case here?

Reply:
I added unit test :)

Reply:
@bamsalem @rszwajko regarding using isDiskNameValid in both create/edit modes:
we can't break any use case since isVmNameValid and isDiskNameValid actually share the same validation logic except length limitation, so now validating of disk alias name in DiskImageEditor is correct.
But two issues just bumped up:
-  is there any reason for using VM_NAME_REGEX.test(nameCandidate) within isVmNameValid function and  /^[a-zA-Z0-9_\-\\.]{1,255}$/.test(nameCandidate) within isDiskNameValid function? We can check the length limitation separately and share the same regex...

- for both DiskImageEditor and Storage.js we use the same validation but display a different error message (pleaseEnterValidDiskName and createVmWizardStorageDiskNameRules). I suggest using the same message.

Reply:
@sgratch  my understanding is that VM_NAME_REGEXP is **less** strict - check [UNICODE_NON_ASCII_LETTERS](https://github.com/oVirt/ovirt-web-ui/blob/a41b92f1e340480129c2cd5f6c9da58066ef96ce/src/components/utils/validation.js#L64)

Reply:
@sgratch
I fixed the validations now we are using only the /^[a-zA-Z0-9_-\.]{1,255}$/.test(nameCandidate) validation.

Regarding the message we display different messages because the message in Create VM wizard must be shorter than DiskImageEditor, that more descriptive:

Long message:
![image](https://user-images.githubusercontent.com/64131213/105989734-35e77480-60aa-11eb-8832-41d87b638372.png)

Short message:
![image](https://user-images.githubusercontent.com/64131213/105990065-a7272780-60aa-11eb-9aa8-059762a719c2.png)

If you still think we should use the same message, WDYT? which one is preferred?